pkg/reconciler/stats_reporter_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"fmt"
+	"testing"
+
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+
+	"knative.dev/pkg/metrics/metricstest"
+)
+
+const (
+	reconcilerMockName    = "mock_reconciler"
+	testResourceNamespace = "test_namespace"
+	testResourceName      = "test_resource"
+)
+
+func TestNewStatsReporter(t *testing.T) {
+	r, err := NewStatsReporter(reconcilerMockName)
+	if err != nil {
+		t.Errorf("Failed to create reporter: %v", err)
+	}
+
+	m := tag.FromContext(r.(*reporter).ctx)
+	v, ok := m.Value(reconcilerTagKey)
+	if !ok {
+		t.Fatalf("Expected tag %q", reconcilerTagKey)
+	}
+	if v != reconcilerMockName {
+		t.Fatalf("Expected %q for tag %q, got %q", reconcilerMockName, reconcilerTagKey, v)
+	}
+}
+
+func TestReporter_ReportKnativeServingEdit(t *testing.T) {
+	reporter, err := NewStatsReporter(reconcilerMockName)
+	if err != nil {
+		t.Errorf("Failed to create reporter: %v", err)
+	}
+	countWas := int64(0)
+	if m := getMetric(t, knativeServingEditCountName); m != nil {
+		countWas = m.Data.(*view.CountData).Value
+	}
+	key := fmt.Sprintf("%s/%s", testResourceNamespace, testResourceName)
+	expectedTags := map[string]string{
+		keyTagKey.Name(): key,
+		reconcilerTagKey.Name(): reconcilerMockName,
+	}
+
+	if err = reporter.ReportKnativeServingEdit(key); err != nil {
+		t.Error(err)
+	}
+	metricstest.CheckCountData(t, knativeServingEditCountName, expectedTags, countWas+1)
+}
+
+func getMetric(t *testing.T, metric string) *view.Row {
```

Comment:
This function can be condensed to (from https://github.com/knative/pkg/blob/master/controller/stats_reporter_test.go)
	initialReconcileCount := int64(0)
	if d, err := view.RetrieveData("reconcile_count"); err == nil && len(d) == 1 {
		initialReconcileCount = d[0].Data.(*view.CountData).Value
	}